tsl/src/continuous_aggs/union_view.c

```diff
@@ -0,0 +1,301 @@
+											 InvalidOid,
+											 InvalidOid,
+											 COERCE_EXPLICIT_CALL);
+
+			break;
+		}
+		case TIMESTAMPOID:
+		{
+			/* timestamp types need to be converted since we store them differently from postgres
+			 * format */
+			Oid converter_oid = LookupFuncName(list_make2(makeString(INTERNAL_SCHEMA_NAME),
+														  makeString(INTERNAL_TO_TS_FUNCTION)),
+											   -1,
+											   NULL,
+											   false);
+			boundary = (Expr *) makeFuncExpr(converter_oid,
+											 partcoltype,
+											 list_make1(boundary),
+											 InvalidOid,
+											 InvalidOid,
+											 COERCE_EXPLICIT_CALL);
+
+			break;
+		}
+		case TIMESTAMPTZOID:
+		{
+			/* timestamp types need to be converted since we store them differently from postgres
+			 * format */
+			Oid converter_oid = LookupFuncName(list_make2(makeString(INTERNAL_SCHEMA_NAME),
+														  makeString(INTERNAL_TO_TSTZ_FUNCTION)),
+											   -1,
+											   NULL,
+											   false);
+			boundary = (Expr *) makeFuncExpr(converter_oid,
+											 partcoltype,
+											 list_make1(boundary),
+											 InvalidOid,
+											 InvalidOid,
+											 COERCE_EXPLICIT_CALL);
+
+			break;
+		}
+
+		default:
+			/* all valid types should be handled above, this should never be reached and error
+			 * handling at earlier stages should catch this */
+			ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg("unsupported datatype for continuous aggregates: %s",
+							format_type_be(partcoltype))));
+			pg_unreachable();
+	}
+
+	CoalesceExpr *coalesce = makeNode(CoalesceExpr);
+	coalesce->coalescetype = partcoltype;
+	coalesce->coalescecollid = InvalidOid;
+	coalesce->args = list_make2(boundary, makeConstLowerBound(partcoltype));
+
+	Expr *op = make_opclause(opno,
+							 BOOLOID,
+							 false,
+							 (Expr *) var,
+							 (Expr *) coalesce,
+							 InvalidOid,
+							 InvalidOid);
+
+	return (Node *) op;
+}
+
+static RangeTblEntry *
+makeRangeTableEntryForSubquery(Query *subquery, const char *aliasname)
+{
+	RangeTblEntry *rte_rel = linitial(subquery->rtable);
+	RangeTblEntry *rte = makeNode(RangeTblEntry);
+	ListCell *lc;
+
+	Assert(list_length(subquery->rtable) == 1);
+
+	rte->rtekind = RTE_SUBQUERY;
+	rte->relid = InvalidOid;
+	rte->subquery = subquery;
+	rte->alias = makeAlias(aliasname, NIL);
+	rte->eref = copyObject(rte->alias);
+
+	foreach (lc, subquery->targetList)
+	{
+		TargetEntry *tle = lfirst(lc);
+		if (!tle->resjunk)
+			rte->eref->colnames = lappend(rte->eref->colnames, makeString(pstrdup(tle->resname)));
+	}
+
+	/*
+	 * Set flags and access permissions.
+	 *
+	 * Subqueries are never checked for access rights.
+	 */
+	rte->lateral = false;
+	rte->inh = false; /* never true for subqueries */
+	rte->inFromCl = true;
+
+	rte->requiredPerms = 0;
+	rte->checkAsUser = InvalidOid;
+	rte->selectedCols = NULL;
+	rte->insertedCols = NULL;
+	rte->updatedCols = NULL;
+
+	return rte;
+}
+
+Query *
```

Comment:
Could you add a comment here describing the transformation  and the union query i.e. mapping a CREATE VIEW stmt to the generated UNION stmt.